src/c++/perf_analyzer/concurrency_manager.cc

```diff
@@ -314,6 +314,12 @@ ConcurrencyManager::Infer(
         while (total_ongoing_requests != 0) {
           std::this_thread::sleep_for(std::chrono::milliseconds(500));
         }
+        // Make sure all threads are in sync with the client's stats
```

Comment:
The issue was that the "end of sequence" requests were not updating the ConcurrencyManager's copy of the client stats (but the client stats were being updated). All of those stats would look like they were part of the first valid request at the next rate.

There were 2 ways to fix this bug. One was to treat all of the "end of sequence" requests as valid (because we sync the stats after every valid request). The other was to add this sync code.  This was much easier to add, so I went with this approach.